src/ragapp/backend/middlewares/rate_limit.py

```diff
@@ -1,37 +1,22 @@
 import os
 import time
 
-import jwt
 from fastapi import HTTPException, Request, status
 from fastapi.responses import Response
-from jwt import InvalidTokenError
-from pydantic import BaseModel, computed_field
 
+from backend.models.jwt import JWT
+from backend.models.user_info import UserInfo
 from backend.services.user_chat_service import UserChatService
 
-JWT_COOKIE_NAME = "Authorization"  # The name of the cookie that stores the JWT token
-JWT_USER_ID_CLAIM = "preferred_username"  # The claim in the JWT token that stores the user ID or user name
-JWT_USER_ROLES_CLAIM = (
-    "X-Forwarded-Roles"  # The claim in the JWT token that stores the user roles
-)
 CHAT_REQUEST_LIMIT_THRESHOLD = int(os.getenv("CHAT_REQUEST_LIMIT_THRESHOLD", 0))
 CHAT_REQUEST_LIMIT_ENABLED = CHAT_REQUEST_LIMIT_THRESHOLD > 0
-ADMIN_ROLE = "admin-manager"
-
-
-class UserInfo(BaseModel):
-    username: str
-    roles: list[str] = []
-
-    @computed_field
-    def is_admin(self) -> bool:
-        return ADMIN_ROLE in self.roles
 
 
 async def request_limit_middleware(request: Request) -> Response:
     if CHAT_REQUEST_LIMIT_ENABLED:
+        jwt = JWT(request.cookies)
+        user = UserInfo.from_jwt_data(jwt.data)
```

Comment:
how about adding a convient function?
```suggestion
        user = UserInfo.from_request(request)
```